### PR TITLE
CAMS-329 Reassign assignments in consolidation

### PR DESCRIPTION
**Pull request by btposey**
# Purpose

Reassign lead case assigned attorneys to child cases.

# Major Changes

* Modify API to reassign child cases when a consolidation is approved.
* Show the child cases and the lead case attorneys in.the consolidation modal.
* Show the currently assigned attorneys for each case in the consolidation accordions.

# Testing/Validation

* Test coverage above threshold.

### CODE REVIEW

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx
Comment (btposey) [inline user-interface/src/data-verification/ConsolidationOrderModal.tsx:65]: We can defer the array copy until we get past the first two length === 0 and length < 3 conditions.

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx
Comment (btposey) [inline user-interface/src/data-verification/ConsolidationOrderModal.tsx:299]: Scrollable DIV

File: user-interface/src/data-verification/ConsolidationOrderAccordion.tsx
Comment (btposey) [inline user-interface/src/data-verification/ConsolidationOrderAccordion.tsx:94]: We could try catch inside the loop.

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx
Comment (btposey) [inline user-interface/src/data-verification/ConsolidationOrderModal.tsx:58]: Need to consider if we can confirm the approval if this API call fails. We should return a 404 for a case that isn't found. In that condition the lead case doesn't exist and as such is invalid to confirm the approval.

Comment (btposey) [reply user-interface/src/data-verification/ConsolidationOrderModal.tsx:59]: Code for this is in branch `CAMS-329-assignment-BRIAN`

Comment (btposey) [reply user-interface/src/data-verification/ConsolidationOrderModal.tsx]: Merged the two branches.

File: user-interface/src/lib/components/uswds/Alert.scss
Comment (btposey) [inline user-interface/src/lib/components/uswds/Alert.scss:50]: Delete

File: user-interface/src/lib/components/uswds/modal/Modal.tsx
Comment (btposey) [inline user-interface/src/lib/components/uswds/modal/Modal.tsx:18]: Remove this since we aren't using it now.

File: user-interface/src/data-verification/ConsolidationCasesTable.tsx
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationCasesTable.tsx:54]: We don't need to do it here, but I think soon we should consider making a Table component to isolate the USWDS classes.

File: user-interface/src/data-verification/ConsolidationCasesTable.tsx
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationCasesTable.tsx:103]: Is `idx` unique enough to use as the key here?

File: user-interface/src/data-verification/ConsolidationOrderAccordion.test.tsx
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderAccordion.test.tsx:299]:
```suggestion
    // click "continue" button on first screen of the modal.
    fireEvent.click(modalApproveButton);
    // click "verify" button on second screen of the modal.
```

The comments above seem more expressive.

File: user-interface/src/data-verification/ConsolidationOrderAccordion.tsx
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderAccordion.tsx:104]: Could we write a named function that would indicate why we are doing these things rather than the comments?

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderModal.tsx:66]: I think a better name might be something like `formatListForDisplay`.

File: user-interface/src/lib/components/uswds/modal/Modal.tsx
Comment (magile32) [inline user-interface/src/lib/components/uswds/modal/Modal.tsx:90]:
```suggestion
    if (closeOnClick) close(e);
```

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderModal.tsx:115]:
```suggestion
    if (!leadCaseSummary || !consolidationType) return; // or should we throw?
    onConfirm({
      status: options.status,
      rejectionReason: reasonRef.current?.value,
      leadCaseSummary: leadCaseSummary,
      consolidationType: consolidationType,
    });
```
I'd prefer to explicitly handle the scenario described in the comments.

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderModal.tsx:189]: Is this something we should have in the Modal component? We'll never want a modal that extends beyond the viewport.

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderModal.tsx:382]: What does the `oxford-comma` class do?

File: user-interface/src/lib/components/LoadingSpinner.tsx
Comment (magile32) [inline user-interface/src/lib/components/LoadingSpinner.tsx:25]:
```suggestion
        id={`loading-spinner-${id}-svg`}
```

Comment (magile32) [reply user-interface/src/lib/components/LoadingSpinner.tsx:25]: Thought it might be useful to denote what the svg is. Unless its being passed thru from the id.

File: user-interface/src/lib/hooks/UseApi.ts
Comment (magile32) [inline user-interface/src/lib/hooks/UseApi.ts:46]: This looks odd to me. Not sure if I understand the responsibility of this GenericApiClient and its implementation compared to the ApiClient.

File: user-interface/src/lib/type-declarations/api.d.ts
Comment (magile32) [inline user-interface/src/lib/type-declarations/api.d.ts:1]: Should these interface be something that belongs in the common module?

File: backend/functions/lib/use-cases/orders/orders.ts
Comment (magile32) [inline backend/functions/lib/use-cases/orders/orders.ts:347]: Recommend a factory pattern here

Comment (jamesobrooks) [reply backend/functions/lib/use-cases/orders/orders.ts:347]: What would that do for us? The factory pattern is best used when dealing with abstract classes with multiple concrete implementations.

File: user-interface/src/lib/components/uswds/modal/SubmitCancelButtonGroup.tsx
Comment (btposey) [reply user-interface/src/lib/components/uswds/modal/SubmitCancelButtonGroup.tsx:12]: It does get defaulted. We cannot set a default value in a type definition.

File: backend/functions/lib/use-cases/orders/orders.ts
Comment (btposey) [reply backend/functions/lib/use-cases/orders/orders.ts:384]: Jon said that was in another ticket.

File: common/src/cams/test-utilities/mock-data.ts
Comment (btposey) [reply common/src/cams/test-utilities/mock-data.ts:254]: I was adding the override pattern we employ in all the other functions.

File: user-interface/src/App.scss
Comment (btposey) [reply user-interface/src/App.scss:35]: Phoenix wanted global vertical alignment on tables in the app.

File: user-interface/src/data-verification/ConsolidationCasesTable.tsx
Comment (btposey) [reply user-interface/src/data-verification/ConsolidationCasesTable.tsx:54]: 💯

File: user-interface/src/lib/type-declarations/api.d.ts
Comment (btposey) [reply user-interface/src/lib/type-declarations/api.d.ts:1]: Yes, we should unify these between the API and UI via common. I just was not prepared to do that big refactor with this ticket since it was so big already.

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx
Comment (btposey) [reply user-interface/src/data-verification/ConsolidationOrderModal.tsx:115]: We should not throw. We have the lead case and the consolidation type as preconditions from the first step. This is handling the second "verify" step. We can put an explicit guard there and even throw just in case we break the logic in step 1 in the future.

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx
Comment (btposey) [reply user-interface/src/data-verification/ConsolidationOrderModal.tsx:189]: Maybe. Probably. Would need to assess how to make this easily generalizable.

File: user-interface/src/lib/components/uswds/modal/SubmitCancelButtonGroup.tsx
Comment (btposey) [reply user-interface/src/lib/components/uswds/modal/SubmitCancelButtonGroup.tsx:12]: Consensus is if we need this complex, multi-view modal interaction again we will create a specialized component for it. For now the tactical solution is good as implemented.

File: user-interface/src/lib/hooks/UseApi.ts
Comment (btposey) [reply user-interface/src/lib/hooks/UseApi.ts:46]: Add some notes about the intention of the `GenericApiClient`.

File: user-interface/src/data-verification/ConsolidationCasesTable.tsx
Comment (btposey) [reply user-interface/src/data-verification/ConsolidationCasesTable.tsx:103]: Probably not.

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx
Comment (btposey) [reply user-interface/src/data-verification/ConsolidationOrderModal.tsx:115]: Loose the comments.

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx
Comment (btposey) [reply user-interface/src/data-verification/ConsolidationOrderModal.tsx:189]: Fine for now. If we encounter this again we need to enhance the Modal component to handle it's own window resize listener hook.

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx
Comment (btposey) [reply user-interface/src/data-verification/ConsolidationOrderModal.tsx:382]: If nothing then remove the class name.

Comment (btposey) [reply user-interface/src/data-verification/ConsolidationOrderModal.tsx:382]: UNLESS this was for testing.

File: user-interface/src/lib/type-declarations/api.d.ts
Comment (btposey) [reply user-interface/src/lib/type-declarations/api.d.ts:1]: Defer to another ticket.

**Review by amorrow-flexion — APPROVED** [review]

**Pull request merged**
